internal/vault/credentials/approle.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package credentials
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	secretsv1alpha1 "github.com/hashicorp/vault-secrets-operator/api/v1alpha1"
+)
+
+type ApproleCredentialProvider struct {
```

Comment:
nit: I think the auth engined is called `AppRole`, may want to rename to `AppRole*`

Reply:
Good catch, thanks!